util/net_help.c

```diff
@@ -285,19 +285,22 @@ int netblockstrtoaddr(const char* str, int port, struct sockaddr_storage* addr,
 }
 
 /* RPZ format address dname to network byte order address */
-static int ipdnametoaddr(uint8_t* dname, struct sockaddr_storage* addr,
-	socklen_t* addrlen, int* af)
+static int ipdnametoaddr(uint8_t* dname, size_t dnamelen,
+	struct sockaddr_storage* addr, socklen_t* addrlen, int* af)
 {
 	uint8_t* ia;
 	size_t dnamelabs = dname_count_labels(dname);
 	uint8_t lablen;
 	char* e = NULL;
 	int z = 0;
+	int len = 0;
 	int i;
 	*af = AF_INET;
-	if(dnamelabs > 6 || dname_has_label(dname, (uint8_t*)"\002zz")) {
+	if(dnamelabs > 6 ||
+		dname_has_label(dname, dnamelen, (uint8_t*)"\002zz")) {
 		*af = AF_INET6;
 	}
+	len = *dname;
 	lablen = *dname++;
```

Comment:
ipdnametoaddr and netblockdnametoaddr also have dnamelen == 0 check needed. And perhaps also check if labellengthbyte fits in dnamelen, and if the label length fits in dnamelen.

Reply:
Added in https://github.com/NLnetLabs/unbound/pull/73/commits/88a706acf8f2af97b86052a2f4b441567d813151

Also subtracting the netblock label (incl. length byte) from length passed to `ipdnametoaddr`.